ivadomed/inference.py

```diff
@@ -336,35 +364,30 @@ def segment_volume(folder_model, fname_images, gpu_id=0, options=None):
     if 'FiLMedUnet' in context and context['FiLMedUnet']['applied']:
         onehotencoder = get_onehotencoder(context, folder_model, options, ds)
         model_params.update({"name": 'FiLMedUnet',
-                        "film_onehotencoder": onehotencoder,
-                        "n_metadata": len([ll for l in onehotencoder.categories_ for ll in l])})
+                             "film_onehotencoder": onehotencoder,
+                             "n_metadata": len([ll for l in onehotencoder.categories_ for ll in l])})
 
     # Data Loader
     data_loader = DataLoader(ds, batch_size=context["training_parameters"]["batch_size"],
                              shuffle=False, pin_memory=True,
                              collate_fn=imed_loader_utils.imed_collate,
                              num_workers=0)
 
-    # MODEL
-    if fname_model.endswith('.pt'):
-        model = torch.load(fname_model, map_location=device)
-        # Inference time
-        model.eval()
-
     # Loop across batches
     preds_list, slice_idx_list = [], []
-
+    last_sample_bool, weight_matrix, volume = False, None, None
     for i_batch, batch in enumerate(data_loader):
-        preds = get_preds(context, model, fname_model, model_params, cuda_available, batch)
+        preds = get_preds(context, path_model_file, model_params, gpu_id, batch)
 
         # Set datatype to gt since prediction should be processed the same way as gt
         for b in batch['input_metadata']:
             for modality in b:
                 modality['data_type'] = 'gt'
 
         # Reconstruct 3D object
-        pred_list, target_list = reconstruct_3d_object(context, batch, undo_transforms, preds, preds_list, kernel_3D, 
-                                                        slice_axis, slice_idx_list, data_loader, fname_images, i_batch)
+        pred_list, target_list, last_sample_bool = reconstruct_3d_object(context, batch, undo_transforms, preds, preds_list, kernel_3D,
```

Comment:
If I try to print `volume` and `weight_matrix` before and after this function, the result is always `None` after, even though they should be passed by reference. If I add a return of `volume` and `weight_matrix`, then they are not None and the test pass.

Reply:
Great point. I will implement that fix tonight and try again. I suspect it might be related to [the issue shown here](https://discuss.pytorch.org/t/what-is-the-recommended-way-to-re-assign-update-values-in-a-variable-or-tensor/6125): i.e. in place update vs object re-assignment of PyTorch tensor objects. Super interesting, super gotcha.